### ISSUE TITLE
KOReader is unable to browse the OPDS catalog provided by Kavita

### ISSUE THREAD

**Issue by andycarlberg** [koreader/koreader#8971]
* KOReader version: 2022.03.1
* Device: Kindle Touch (K5)

#### Issue

I am running a Kareadita/Kavita instance on my home server with the OPDS API enabled. I am not able to brows the catalog from KOReader. I was able to connect and can view the catalog root but I can't go beyond the catalog root.

I receive one of three messages

1. "Failed to parse catalog"
2. "Server refuses to serve uncompressed content"
3. 500 Server error

I've opened an issue on the Kavita repo as well since I don't know where the problem actually is.

Kareadita/Kavita#1199

#### Steps to reproduce

1. On a device running KOReader, go to the OPDS catalog section
2. Enter the Kavita catalog URL for your user
3. Save
4. Browse the OPDS catalog
5. See one of the three errors listed above

**Comment by NiLuJe** [comment]
I think we added some logging around these, so a verbose debug log would be helpful ;).

**Comment by andycarlberg** [comment]
Sorry, I'm new to KOReader. Can you point me to some instructions on viewing and providing those logs?

**Comment by NiLuJe** [comment]
In the File manager's top menu > Tools > More tools > Developer options;  tick both debug & verbose logging and restart KOReader.

Then check/post the crash.log file in the koreader folder.

**Comment by andycarlberg** [comment]
Here's the full log file. I attempted to find the portions that are relevant but it seemed to be the majority of the file. If there's something I can point to, please let me know
[crash.log](https://github.com/koreader/koreader/files/8421207/crash.log)
.

**Comment by NiLuJe** [comment]
Err, is there actually a failure in your log?

**Comment by andycarlberg** [comment]
Well this is awkward. I did see the errors I mentioned when creating the log I posted. In trying to create a new, cleaner log just now, I didn't receive the errors and everything worked as expected. I don't think I did anything different. I did regenerate the API key provided by Kavita since I had publicly posted the original in the log file. That's the only change that I can think of.

I guess we can consider this resolved. If I see the issue again I will report back. Sorry for the trouble and thanks for the quick responses! Looking forward to getting more out of my Kindle now!
